lib/record_store/provider/ns1.rb

```diff
@@ -0,0 +1,192 @@
+require 'ns1'
+
+module RecordStore
+  class Provider::NS1 < Provider
+    class << self
+      def client
+        Provider::NS1::Client.new(api_key: secrets['api_key'])
+      end
+
+      # Downloads all the records from the provider.
+      #
+      # Returns: an array of `Record` for each record in the provider's zone
+      def retrieve_current_records(zone:, stdout: $stdout)
+        records_for_zone(zone)
+         .map{|short_record| build_from_api(short_record, zone)}
```

Comment:
Style: the spacing does not seem right here
Could we add a linter to this project to help everyone writing code in it ? https://rubocop.readthedocs.io/en/stable/ could be a solution (it's OSS, and we use it for other internal projects, not sure how complicated it would be to setup for that public project, but pretty sure it's doable)

Reply:
It has been updated as you suggested, thanks :)